trin-execution/src/execution.rs

```diff
@@ -156,7 +75,7 @@ impl TrinExecution {
             self.node_data_directory.clone(),
         );
         let range_start = Instant::now();
-        let mut last_executed_block_number = start - 1;
+        let mut last_executed_block_number = u64::MAX;
```

Comment:
nit: I think with the check at line 66 and most recent changes, you can do what I proposed few reviews ago regarding keeping track of the last executed block (instead of `last_executed_block_number` and `last_state_root`).

Something like this:

```rust
let mut last_executed_block_header: Option<Header> = None;
for block_number in start..=end {
    ...
    block_executor.execute_block(&block)?;
    last_executed_block_header = Some(block.header);

    ...
}

let last_executed_block_header = last_executed_block_header.expect("At least one block must have been executed");

let root_with_trie_diff = block_executor.commit_bundle()?;
ensure!(root_with_trie_diff.root == last_executed_block_header.state_root, ... );

self.execute_position.update(self.database.db.clone(), last_executed_block_header);
```

Up to you.